### DISCUSSION TITLE
The inferred type of 'client' cannot be named without a reference to '../prisma/node_modules/@prisma/client/runtime'

### DISCUSSION THREAD

**Discussion by michealroberts** [trpc/trpc#2261] [Q&A] [answered]
I'm currently using Prisma with tRPC.io in a turborepo monorepo and I am seeing the following error on build:

```
The inferred type of 'client' cannot be named without a reference to '../prisma/node_modules/@prisma/client/runtime'. This is likely not portable. A type annotation is necessary.
```

<img width="1047" alt="Screenshot 2022-07-19 at 16 44 01" src="https://user-images.githubusercontent.com/84131395/179792636-ee6bcb19-5afb-4c23-a093-b36ae990cd6a.png">

I'm using version 9.26.2 on both @trpc/server and @trpc/client ...

I have generated the prisma types ... but no luck, I still see this error on build.

I can provide access to the monorepo in question for reference ... for a replication of the error.

**Comment by jlalmes** [comment, marked as answer]
Could you share the reproduction and I'll take look!

**Reply by michealroberts** [reply]
@jlalmes I have sent an invite to the private repo ! :)

**Reply by jlalmes** [reply]
This appears to be a TS error related to `pnpm` (https://github.com/microsoft/TypeScript/issues/42873). I was able to resolve the issue by removing `"declaration": true,` from `tsconfig.json` - PR submitted.

**Reply by michealroberts** [reply]
Yes, that does fix the issue ... I guess the idea is you shouldn't really have your AppRouter as an exported member within a package. It has to be a mono repo style setup or within the same project?

**Reply by michealroberts** [reply]
Also, could you link the PR please 🙏

**Reply by jlalmes** [reply]
https://github.com/observerly/sirius/issues/2
